### PR TITLE
feature: add HuggingFace model and predictor

### PR DESCRIPTION
**Pull request by ahsan-z-khan**
*Issue #, if available:*

*Description of changes:*
created model.py for HF. updated image uris and integ test.
*Testing done:*
unit tests
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by apogupta2018 — DISMISSED** [review]
Merge pending tests passing.

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -1114,7 +1114,7 @@ def prepare_container_def(self, instance_type=None, accelerator_type=None):
     def _upload_code(self, key_prefix, repack=False):
         """Placeholder Docstring"""
         local_code = utils.get_config_value("local.local_code", self.sagemaker_session.config)
-        if self.sagemaker_session.local_mode and local_code:
+        if (self.sagemaker_session.local_mode and local_code) or self.entry_point is None:
```
Comment (vdantu) [inline src/sagemaker/model.py:1117]: What's this code doing? If a customer uses a training script whole training, would this entry_point be None when we do estimator.deploy

File: src/sagemaker/model.py [hunk 2]
```diff
@@ -1127,7 +1127,7 @@ def _upload_code(self, key_prefix, repack=False):
                 dependencies=self.dependencies,
             )
 
-        if repack:
+        if repack and self.model_data is not None and self.entry_point is not None:
```
Comment (vdantu) [inline src/sagemaker/model.py:1130]: If we just run estimator.fit followed by estimator.deploy, would this block of code inside the if statement be executed?

**Review by apogupta2018 — DISMISSED** [review]

File: tests/integ/test_huggingface.py [hunk 3]
```diff
@@ -104,3 +106,37 @@ def test_huggingface_training_tf(
...
+        "HF_MODEL_ID": "sshleifer/tiny-distilbert-base-uncased-finetuned-sst-2-english",
+        "HF_TASK": "text-classification",
+    }
+    endpoint_name = unique_name_from_base("test-hf-inference")
+
+    model = HuggingFaceModel(
+        sagemaker_session=sagemaker_session,
+        role="SageMakerRole",
+        env=env,
+        image_uri="763104351884.dkr.ecr.us-west-2.amazonaws.com/huggingface-tensorflow-inference:2.4.1-gpu",
```
Comment (saimidu) [inline tests/integ/test_huggingface.py:129]: This doesn't need to be hardcoded anymore.

Comment (ahsan-z-khan) [reply tests/integ/test_huggingface.py:129]: right. But i tried on with framework args. Gives me this error:
`The image '763104351884.dkr.ecr.us-east-2.amazonaws.com/huggingface-pytorch-inference:1.7.1-transformers4.6.1-gpu-py36' does not exist..`

**Pull request closed without merging**